src/encoding.iced

```diff
@@ -29,14 +29,15 @@ exports.Encoding = class Encoding
     encoded_bits - decoded_bits
 
   # encode a block of length @in_block_len or less....
-  encode_block : (block) ->
+  encode_block : (block, old_shift) ->
     # This is the minimal number of encoding bytes it takes to
     # output the input block
     encoded_len = @encoded_len block.length
     # Left shift away all bits that are always going to be 0
     shift = @extra_bits {encoded_len, decoded_len : block.length }
     # The raw big-endian representation of the block
-    num = (nbi().fromBuffer block).shiftLeft shift
+    num = (nbi().fromBuffer block)
+    if old_shift then num = num.shiftLeft(shift)
```

Comment:
Go ahead and include the whole `@extra_bits` computation inside the `if`. There's a logarithm somewhere in there that might not be cheap compared to all the other operations here.